ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/[slug]/runs/page.tsx

```diff
@@ -51,14 +65,35 @@ export default function RunsPage() {
   const [rawFilteredStatus, setFilteredStatus, removeFilteredStatus] =
     useStringArraySearchParam('filterStatus');
   const [rawTimeField, setTimeField] = useSearchParam('timeField');
+  const [lastDays, setLastDays] = useSearchParam('last');
+
+  /* TODO: Time params for absolute time filter */
+  // const [fromTime, setFromTime] = useSearchParam('from');
+  // const [untilTime, setUntilTime] = useSearchParam('until');
+
+  /* TODO: When we have absolute time, the start date will be either coming from the date picker or the relative time */
+  const [startTime, setStartTime] = useState<Date>(new Date());
+
+  useEffect(() => {
+    if (lastDays) {
+      setStartTime(
+        getTimestampDaysAgo({
+          currentDate: new Date(),
+          days: parseInt(lastDays),
+        })
+      );
+    } else {
+      setStartTime(TimeRangeFilterDefault.startTime);
```

Comment:
This is gonna get stale after the page loads. Probably an esoteric edge case, but it's still cleaner to avoid this `else` block by defaulting `lastDays` if it isn't set:

```ts
const [lastDays = "3", setLastDays] = useSearchParam('last')
```